scapy/arch/linux.py

```diff
@@ -146,17 +149,20 @@ def attach_filter(s, bpf_filter, iface):
     if f.close():
         raise Scapy_Exception("Filter parse error")
     nb = int(lines[0])
-    bpf = ""
+    bpf = b""
     for l in lines[1:]:
         bpf += struct.pack("HBBI", *(int(e) for e in l.split()))
 
-    # XXX. Argl! We need to give the kernel a pointer on the BPF,
-    # python object header seems to be 20 bytes. 36 bytes for x86 64bits arch.
-    bpfh = struct.pack("HL", nb, id(bpf) + (36 if IS_64BITS else 20))
+    # XXX. Argl! We need to give the kernel a pointer on the BPF
```

Comment:
passing the pointer through `id()` isn't accepted anymore on python 3, or at least we can't make it python 2 compatible